### PR TITLE
Adds technical metadata generator service/job and dro model

### LINKED ISSUE #3: Setup database to hold technical metadata

### LINKED ISSUE #9: Create a background job
## In the background job:
Assume a table with compound key (druid,filename) and with a md5 column:
Calculate the md5s of each provided file
  - For each (druid, filename, md5) match, no need to recalculate technical metadata.
  - For any md5s not calculated from the provided file list, but that exist in the database for the druid,  filename match delete them.
  - For any new filenames or md5, recalculate the tech md, insert. 

Once all files have been recorded, notify the workflow service.

### LINKED ISSUE #11: Create a metadata record model
## Fields
- druid
- filename
- md5
- byte size
- filetype (mime?)
- created_at
- updated_at
- tool_versions (sigfried version, exiftool version, etc)

### PR DESCRIPTION
**Pull request by justinlittman**
Adds a technical metadata generator service and job. Also, adds a dro file model.

closes #9. closes #11. closes #3.

### CODE REVIEW

File: app/services/technical_metadata_generator.rb [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      druid: druid,
+      filename: filename_for(filepath),
+      md5: md5,
+      bytes: ::File.size(filepath),
+      created_at: Time.zone.now,
+      updated_at: Time.zone.now
+    }
+  end
+
+  def dro_files_to_delete
```
Comment (mjgiarlo) [inline app/services/technical_metadata_generator.rb]: If this method returns an `ActiveRecord::Relation`, we'll be able to drop the iteration on line 37 and go with `to_delete.destroy_all`.

Worth looking into collapsing the filename check into the `#where` call? Thinking here about pushing as much into the DB within reason.

File: app/services/technical_metadata_generator.rb [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    # Generate technical metadata for each file. If any errors, fail.
+    filepaths.each { |filepath| generate_for_file(filepath) }
+    return errors unless errors.empty?
+
+    # Find DroFiles that should be deleted.
+    to_delete = dro_files_to_delete
+
+    ApplicationRecord.transaction do
+      to_delete.each(&:destroy)
+      DroFile.upsert_all(upserts, unique_by: %i[druid filename])
```
Comment (mjgiarlo) [inline app/services/technical_metadata_generator.rb]: Dig the usage of `#upsert_all` here! TIL.

**Comment by mjgiarlo** [comment]
@justinlittman Linked this PR to #3.

**Comment by justinlittman** [comment]
@mjgiarlo Switched to using jsonb. Ready for re-review.

**Pull request merged**